jdisc_http_service/src/main/java/com/yahoo/jdisc/http/filter/DiscFilterRequest.java

```diff
@@ -100,12 +91,17 @@ public void setRemoteAddr(String remoteIpAddress) {
      *  on which the request was received.
      */
     public String getLocalAddr() {
-        if (null == localAddress.getAddress()) {
-            return null;
-        }
+        InetSocketAddress localAddress = localAddress();
+        if (localAddress.getAddress() == null) return null;
         return localAddress.getAddress().getHostAddress();
     }
 
+    protected InetSocketAddress localAddress() {
```

Comment:
I had prefer making this deprecated or removing it altogether. Any future users of this method may not know about the implicit DNS lookup. The method looks like a getter but may actually do IO, which is bad design.

Reply:
My reasoning was that the field was protected for a reason, but ok.